### PR TITLE
Matrix chain multiplication

### PR DESCRIPTION
**Pull request by smriti1313**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
yes

### Description

added matrix chain multiplication algorithm using dynamic programming

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue: #2094 
pull request #2123

### CODE REVIEW

File: Python/dp/matrix_chain_multiplication.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    else:
+        s = "("
+        k = S[i][j]
+        s += print_matrix_chain(S, i, k)
+        s += print_matrix_chain(S, k + 1, j)
+        s += ")"
+        return s
+
+t, s = matrix_chain_order(arr)
+print(print_matrix_chain(s, 1, 3))
```
Comment (HarshCasper) [inline Python/dp/matrix_chain_multiplication.py:85]: Add this in a `if __name__ == '__main__'` block

File: Python/dp/matrix_chain_multiplication.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+def print_matrix_chain(S, i, j):
+    if i == j:
+        return " A_" + str(i) + " "
+    else:
+        s = "("
+        k = S[i][j]
+        s += print_matrix_chain(S, i, k)
+        s += print_matrix_chain(S, k + 1, j)
+        s += ")"
+        return s
```
Comment (HarshCasper) [inline Python/dp/matrix_chain_multiplication.py:82]: Use meaningful variable names

File: Python/dp/matrix_chain_multiplication.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+
+arr = [5,2,4,9,1]
+print(chain_matrix_multiplication(arr, 1, len(arr) - 1))
+
+
+def matrix(m, n):
+    return [[0] * n for _ in range(m)]
+
+def matrix_chain_order(p):
```
Comment (HarshCasper) [inline Python/dp/matrix_chain_multiplication.py:50]: Add function docstring

File: Python/dp/matrix_chain_multiplication.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+        k += 1
+    return cMin
+
+
+arr = [5,2,4,9,1]
+print(chain_matrix_multiplication(arr, 1, len(arr) - 1))
+
+
+def matrix(m, n):
```
Comment (HarshCasper) [inline Python/dp/matrix_chain_multiplication.py:47]: Add function docstring

File: Python/dp/matrix_chain_multiplication.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+Time Complexity: O(n^3)
+Auxiliary Space: O(n^2)
+
+"""
+
+
+import sys
+
+def chain_matrix_multiplication(matrices, i, j):
```
Comment (HarshCasper) [inline Python/dp/matrix_chain_multiplication.py:26]: Add function docstring

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Please revert the README changes. 

We can have a separate issue for that

**Comment by atarax665** [comment]
No update!

**Pull request closed without merging**